sound/soc/sof/intel/bdw.c

```diff
@@ -518,13 +519,10 @@ static int bdw_fw_ready(struct snd_sof_dev *sdev, u32 msg_id)
 				 fw_ready->hostbox_offset,
 				 fw_ready->hostbox_size);
 
-	dev_info(sdev->dev,
-		 " Firmware info: version %d:%d-%s build %d on %s:%s\n",
-		 v->major, v->minor, v->tag, v->build, v->date, v->time);
-
-	/* only copy the fw_version into debugfs at first boot */
-	if (sdev->first_boot)
-		memcpy(&sdev->fw_version, v, sizeof(*v));
+	/* make sure ABI version is compatible */
+	ret = snd_sof_ipc_valid(sdev);
+	if (ret < 0)
+		return ret;
```

Comment:
shouldn't we check this only for the first boot, instead of checking at every resume?

Reply:
This is just a copy of existing flow, but I have more to come. See below.